python/cudf/cudf/io/csv.py

```diff
@@ -533,11 +558,10 @@ def _get_plc_data_type_from_dtype(dtype) -> plc.DataType:
     # https://github.com/rapidsai/cudf/issues/3960
     if isinstance(dtype, cudf.CategoricalDtype):
         dtype = dtype.categories.dtype
-    elif dtype == "category":
-        dtype = "str"
-
-    if isinstance(dtype, str):
-        if dtype == "date32":
+    elif isinstance(dtype, str):
+        if dtype == "category":
+            return plc.DataType(plc.types.TypeId.STRING)
```

Comment:
Don't we have a constant for this somewhere now?

Reply:
And if so, should we have similar constants for the other dtypes below like timestamps?

Reply:
I realized I could remove all this logic since we are guaranteed a dtype object here (and not strings), so we can use an existing helper function instead

Reply:
Nice that's even better.